lib/que/adapters/yugabyte.rb

```diff
@@ -0,0 +1,75 @@
+# frozen_string_literal: true
+
+# https://github.com/que-rb/que/blob/80d6067861a41766c3adb7e29b230ce93d94c8a4/lib/que/active_job/extensions.rb
+module Que
+    module Adapters
+      class Yugabyte < Que::Adapters::ActiveRecord
+        def initialize
+            super
+        end
+
+        def checkout_activerecord_adapter(&block)
+            YugabyteRecord.connection_pool.with_connection(&block)
+        end
+
+        def checkout_lock_database_connection
+            # when multiple threads are running we need to make sure
+            # the acquiring and releasing of advisory locks is done by the 
+            # same connection
+            Thread.current[:db_connection] ||= LockDatabaseRecord.connection_pool.checkout
+        end
+
+        def lock_database_connection
+            Thread.current[:db_connection]
+        end
+
+        def release_lock_database_connection
+            LockDatabaseRecord.connection_pool.checkin(Thread.current[:db_connection])
+        end
+
+        def execute(command, params=[])
+            if command == :lock_job
+                queue, cursor = params
+                lock_job_with_lock_database(queue, cursor)
+            elsif command == :unlock_job
+                job_id = params[0]
+                unlock_job(job_id)
+            else
+                super(command, params)
+            end
+        end
+
+        def lock_job_with_lock_database(queue, cursor)
+            query = QueJob.select(:job_id, :queue, :priority, :run_at, :job_class, :retryable, :args, :error_count)
+                        .select("extract(epoch from (now() - run_at)) as latency")
+                        .where("queue = ? AND job_id >= ? AND run_at <= ?", queue, cursor, Time.now)
+                        .where(retryable: true)
+                        .order(:priority, :run_at, :job_id)
+                        .limit(1).to_sql
+
+            result = Que.execute(query)
+            return result if result.empty?
+         
+             if locked?(result.first['job_id'])
+                return result
+             end
```

Comment:
An idea I had. We can do something like this to make this a bit more efficient, basically we use the row itself as a temporary lock to prevent a lot of race conditions. This way, a lot of the work is done within the main database itself.

Let me know what you think.
```ruby
loop do
  job_connection.transaction do
    # This locks the row only while we are checking the advisory lock database.
    # This saves duplicate attempts by different works to acquire the lock.
    result = Que.execute("SELECT ... FOR UPDATE SKIPPED LOCK")

    return result if result.empty?

    if locked?(result.first['job_id'])
      return result
    end
    # lock is released after the advisory lock check as the transaction completes
  end
end
```

Reply:
we will have cross database transaction in this case. If that's not an issue then this would be good

Reply:
> we will have cross database transaction in this case

Cross database transactions mainly a problem when you're trying to maintain consistency of writes between two databases (eg. if one commits but the other rolls back).

I don't think this is a problem here: technically acquiring the a lock _is_ a write, but the `SELECT FOR UPDATE` is temporary and not the source of truth, and the lock is released at the end of the transaction anyway.

Reply:
tried this, but for some reason, all the specs are getting hanged. It would require more time to debug what's going on. Will address this separate PR